code/go/0chain.net/blobbercore/handler/grpc_handler_integration_test.go

```diff
@@ -908,6 +909,394 @@ func TestBlobberGRPCService_IntegrationTest(t *testing.T) {
+		allocationTx := randString(32)
+
+		pubKey, _, signScheme := GeneratePubPrivateKey(t)
+		clientSignature, _ := signScheme.Sign(encryption.Hash(allocationTx))
+		pubKeyBytes, _ := hex.DecodeString(pubKey)
+		clientId := encryption.Hash(pubKeyBytes)
+
+		if err := tdController.ClearDatabase(); err != nil {
+			t.Fatal(err)
+		}
+		if err := tdController.AddRenameTestData(allocationTx, pubKey, clientId); err != nil {
+			t.Fatal(err)
+		}
+
+		testCases := []struct {
+			name            string
+			context         metadata.MD
+			input           *blobbergrpc.RenameObjectRequest
+			expectedMessage string
+			expectingError  bool
+		}{
+			{
+				name: "Success",
+				context: metadata.New(map[string]string{
+					common.ClientHeader:          clientId,
+					common.ClientSignatureHeader: clientSignature,
+					common.ClientKeyHeader:       pubKey,
+				}),
+				input: &blobbergrpc.RenameObjectRequest{
+					Allocation:   allocationTx,
+					Path:         "/some_file",
+					PathHash:     "exampleId:examplePath",
+					ConnectionId: "connection_id",
+					NewName:      "some_new_file",
+				},
+				expectedMessage: "some_new_file",
+				expectingError:  false,
+			},
+			{
+				name: "Fail",
+				context: metadata.New(map[string]string{
+					common.ClientHeader:          clientId,
+					common.ClientSignatureHeader: clientSignature,
+					common.ClientKeyHeader:       pubKey,
+				}),
+				input: &blobbergrpc.RenameObjectRequest{
+					Allocation:   "",
+					Path:         "",
+					PathHash:     "",
+					ConnectionId: "",
+					NewName:      "",
+				},
+				expectedMessage: "",
+				expectingError:  true,
+			},
+		}
+
+		for _, tc := range testCases {
+			ctx := context.Background()
+			ctx = metadata.NewOutgoingContext(ctx, tc.context)
+			response, err := blobberClient.RenameObject(ctx, tc.input)
+			if err != nil {
+				if !tc.expectingError {
+					t.Fatal(err)
+				}
+
+				continue
+			}
+
+			if tc.expectingError {
+				t.Fatal("expected error")
+			}
+
+			if response.GetFilename() != tc.expectedMessage {
+				t.Fatal("failed!")
+			}
+		}
+	})
+
+	t.Run("TestDownload", func(t *testing.T) {
+		allocationTx := randString(32)
+
+		root, _ := os.Getwd()
+		path := strings.Split(root, `code`)
+
+		f, err := os.Create(path[0] + `docker.local/blobber1/files/files/exa/mpl/eId/objects/tmp/Mon/Wen/MyFile`)
+		if err != nil {
+			t.Fatal(err)
+		}
+		file, err := os.Open(root + "/grpc_handler_integration_test.go")
+		if err != nil {
+			t.Fatal(err)
+		}
+		stat, err := file.Stat()
+		if err != nil {
+			t.Fatal(err)
+		}
+		fileB := make([]byte, stat.Size())
+		if _, err := io.ReadFull(file, fileB); err != nil {
+			t.Fatal(err)
+		}
+		_, err = f.Write(fileB)
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = f.Write(fileB)
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = f.Write(fileB)
+		if err != nil {
+			t.Fatal(err)
+		}
+		fmt.Println(`file size`, stat.Size())
+		file.Close()
+		f.Close()
```

Comment:
why did you do this, where is this being used please show me in PR

Reply:
Noted

Reply:
Wasnt talking about println but I think you answered this in previous comment
```
it is mounted as volume in b0docker-compose.yml. And passed it as a flag. In main.go, it ie set to FsStore. line 134 in main.go
```